logfire/_internal/config.py

```diff
@@ -592,8 +613,15 @@ def _initialize(self) -> ProxyTracerProvider:
             # Both recommend generating a UUID.
             resource = Resource({ResourceAttributes.SERVICE_INSTANCE_ID: uuid4().hex}).merge(resource)
 
+            # Avoid using the usual sampler if we're using tail-based sampling.
+            # The TailSamplingProcessor will handle the random sampling part as well.
+            sampler = (
+                ParentBasedTraceIdRatio(self.trace_sample_rate)
+                if self.trace_sample_rate < 1 and self.tail_sampling is None
+                else None
+            )
```

Comment:
How do we document / tell users they can't combine these? Is there a world where I want to head sample down to 10% (to reduce overhead in the SDK) and then tail sample down to 1%? There's still advantages to head sampling.

Reply:
They can combine them, see the PR body or the new `test_random_sampling`. 

> Is there a world where I want to head sample down to 10% (to reduce overhead in the SDK) and then tail sample down to 1%?

I don't know what this means. Tail sample down to a percentage?

It sounds like you want to be able to discard most spans up front randomly regardless of whether tail-sampling would include them, so that a span only gets through if it's 'notable' AND 'lucky'.

Reply:
Yes exactly